lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <map>
+
+#include "ShardTracker.h"
+#define MAX_SHARD_SIZE 100
+
+namespace magma {
+
+ShardTracker::ShardTracker() : number_of_shards_(0) {}
+
+int ShardTracker::add_ue() {
+  int shard_id;
+  // If we have no shards, we add an entry for the shard representing
+  // of 1 representing a single UE at the new shard*/
+  if (shards_.size() == 0) {
+    shard_id = 0;
+    shards_.push_back(1);
+    number_of_shards_++;
+  } else {
+    // We iterate through all existing shards, if all are full
+    // we create a new shard with quantity 1, otherwise we increment
+    // the quantity of UEs in the latest shard
+    for (auto i = 0; i < number_of_shards_; i++) {
```

Comment:
stylistic -> for (auto& shard: shards_)

Reply:
Are for range loops discouraged(didn't see this mentioned in the style guide). The index is actually needed to return the relevant shard id(mentioned in ShardTracker.h), so wouldn't changing up the for loop form just make for more code?

Reply:
ah right you need the offset index, missed that.